File: snapaccount-py/routes/health.py


from fastapi import APIRouter
from config import API_VERSION, PAYSTACK_SECRET_KEY
from services.ocr import is_reader_ready
from services.bank_registry import get_cache_info

router = APIRouter()

@router.get("/health")
async def health():
    """Service health check — no auth required."""
    cache_info = get_cache_info()

    return {
        "status": "healthy",
        "version": API_VERSION,
        "dependencies": {
            "ocr_engine": "ready" if is_reader_ready() else "not_loaded",
            "name_enquiry": (
                "configured"
                if PAYSTACK_SECRET_KEY and PAYSTACK_SECRET_KEY != "sk_test_xxxx"
                else "not_configured"
            ),
            "bank_registry": "healthy" if cache_info["total_count"] > 0 else "empty",
        },
        "bank_registry": cache_info,
    }

File: src/routes/health.ts

import { Router, Request, Response } from 'express';
import { config } from '../config';
import { getCacheInfo } from '../services/bankRegistry';

const router = Router();

router.get('/health', (_req: Request, res: Response) => {
  const bankCache = getCacheInfo();

  res.json({
    status: 'healthy',
    version: config.apiVersion,
    timestamp: new Date().toISOString(),
    dependencies: {
      ocr_engine: 'healthy',
      name_enquiry: config.paystackSecretKey ? 'configured' : 'not_configured',
      bank_registry: bankCache.totalCount > 0 ? 'healthy' : 'empty',
    },
    bank_registry: {
      total_banks: bankCache.totalCount,
      last_updated: bankCache.lastUpdated,
    },
  });
});

export default router;
